components/ulp/include/esp32/ulp.h

```diff
@@ -46,50 +46,62 @@ extern "C" {
+#define OPCODE_WR_REG 1         /*!< Instruction: write peripheral register (RTC_CNTL/RTC_IO/SARADC) */
 
-#define OPCODE_RD_REG 2         /*!< Instruction: read peripheral register (RTC_CNTL/RTC_IO/SARADC) (not implemented yet) */
+#define OPCODE_RD_REG 2         /*!< Instruction: read peripheral register (RTC_CNTL/RTC_IO/SARADC) */
 
 #define RD_REG_PERIPH_RTC_CNTL 0    /*!< Identifier of RTC_CNTL peripheral for RD_REG and WR_REG instructions */
 #define RD_REG_PERIPH_RTC_IO   1    /*!< Identifier of RTC_IO peripheral for RD_REG and WR_REG instructions */
 #define RD_REG_PERIPH_SENS     2    /*!< Identifier of SARADC peripheral for RD_REG and WR_REG instructions */
 #define RD_REG_PERIPH_RTC_I2C  3    /*!< Identifier of RTC_I2C peripheral for RD_REG and WR_REG instructions */
 
-#define OPCODE_I2C 3            /*!< Instruction: read/write I2C (not implemented yet) */
+#define OPCODE_I2C 3            /*!< Instruction: read/write I2C */
+#define OPCODE_I2C_RD 0         /*!< I2C read */
+#define OPCODE_I2C_WR 1         /*!< I2C write */
 
 #define OPCODE_DELAY 4          /*!< Instruction: delay (nop) for a given number of cycles */
 
-#define OPCODE_ADC 5            /*!< Instruction: SAR ADC measurement (not implemented yet) */
+#define OPCODE_ADC 5            /*!< Instruction: SAR ADC measurement */
 
 #define OPCODE_ST 6             /*!< Instruction: store indirect to RTC memory */
 #define SUB_OPCODE_ST 4         /*!< Store 32 bits, 16 MSBs contain PC, 16 LSBs contain value from source register */
 
 #define OPCODE_ALU 7            /*!< Arithmetic instructions */
 #define SUB_OPCODE_ALU_REG 0    /*!< Arithmetic instruction, both source values are in register */
 #define SUB_OPCODE_ALU_IMM 1    /*!< Arithmetic instruction, one source value is an immediate */
-#define SUB_OPCODE_ALU_CNT 2    /*!< Arithmetic instruction between counter register and an immediate (not implemented yet)*/
+#define SUB_OPCODE_ALU_CNT 2    /*!< Arithmetic instruction between counter register and an immediate */
 #define ALU_SEL_ADD 0           /*!< Addition */
 #define ALU_SEL_SUB 1           /*!< Subtraction */
 #define ALU_SEL_AND 2           /*!< Logical AND */
 #define ALU_SEL_OR  3           /*!< Logical OR */
 #define ALU_SEL_MOV 4           /*!< Copy value (immediate to destination register or source register to destination register */
 #define ALU_SEL_LSH 5           /*!< Shift left by given number of bits */
 #define ALU_SEL_RSH 6           /*!< Shift right by given number of bits */
+#define ALU_SEL_SINC  0         /*!< Increment counter register */
+#define ALU_SEL_SDEC  1         /*!< Decrement counter register */
+#define ALU_SEL_SRST  2         /*!< Reset counter register */
 
 #define OPCODE_BRANCH 8         /*!< Branch instructions */
 #define SUB_OPCODE_BX  0        /*!< Branch to absolute PC (immediate or in register) */
+#define SUB_OPCODE_BR  1        /*!< Branch to relative PC */
+#define SUB_OPCODE_BS  2        /*!< Branch to relative PC */
```

Comment:
Could you please add ", conditional on R0" and ", conditional on the stage counter" to these two comments?